host/sgx/loadelf.c

```diff
@@ -224,9 +224,10 @@ static oe_result_t _initialize_image_segments(
             case PT_LOAD:
                 if (lo > ph->p_vaddr)
                     lo = ph->p_vaddr;
-
-                if (hi < ph->p_vaddr + ph->p_memsz)
-                    hi = ph->p_vaddr + ph->p_memsz;
+                uint64_t temp_hi = hi;
+                OE_CHECK(oe_safe_add_u64(ph->p_vaddr, ph->p_memsz, &temp_hi));
+                if (hi < temp_hi)
+                    hi = temp_hi;
```

Comment:
I think you can just pass hi into oe_safe_add. If there is an overflow, it will fail.

Reply:
hi is updated conditionally only if new high is greater than previous  hi. o i think temp variable is required.